libs/core/datastructures/include/hpx/datastructures/tuple.hpp

```diff
@@ -503,31 +503,34 @@ namespace hpx {
 
     // template <size_t I, class Tuple>
     // class tuple_element
-    template <std::size_t I, typename T>
+    template <std::size_t I, typename T, typename Enable>
     struct tuple_element
     {
     };
 
     template <std::size_t I, typename T>
-    struct tuple_element<I, const T>
+    struct tuple_element<I, const T,
+        typename util::always_void<typename tuple_element<I, T>::type>::type>
       : std::add_const<typename tuple_element<I, T>::type>
     {
     };
 
     template <std::size_t I, typename T>
-    struct tuple_element<I, volatile T>
+    struct tuple_element<I, volatile T,
+        typename util::always_void<typename tuple_element<I, T>::type>::type>
       : std::add_volatile<typename tuple_element<I, T>::type>
     {
     };
 
     template <std::size_t I, typename T>
-    struct tuple_element<I, const volatile T>
+    struct tuple_element<I, const volatile T,
+        typename util::always_void<typename tuple_element<I, T>::type>::type>
       : std::add_cv<typename tuple_element<I, T>::type>
     {
     };
 
     template <std::size_t I, typename... Ts>
-    struct tuple_element<I, tuple<Ts...>>
+    struct tuple_element<I, tuple<Ts...>, void>
```

Comment:
Do we need the explicit `, void>` ?

Reply:
No, I don't think so. Good catch. I've defaulted it (but not tested it).